cloud/clouds.go

```diff
@@ -156,6 +156,7 @@ type cloudSet struct {
 
 // cloud is equivalent to Cloud, for marshalling and unmarshalling.
 type cloud struct {
+	Name             string                 `yaml:"name"`
```

Comment:
We're serialising out the cloud name here now - does that mean that we'll do the wrong thing for users that have exisiting yaml files without the name attribute in clouds? I guess it'll be ok as long as the reading still code overrides the name from the key. Although that does make it possible for the key and the name to be inconsistent in the file (which is why I didn't change this when I added name). Is that a problem? It might be confusing.